lib/core/src/Cardano/DB/Sqlite.hs

```diff
@@ -372,6 +372,7 @@ data DBLog
     | MsgWaitingForDatabase Text (Maybe Int)
     | MsgRemovingInUse Text Int
     | MsgRemoving Text
+    | MsgRemovingDatabaseEntity Text
```

Comment:
How about being more specific?
`MsgRemovePool PoolId`

Reply:
> How about being more specific?
> `MsgRemovePool PoolId`

Because the `DBLog` type is a general type, shared by the wallet and the pool databases.

As far as I can tell, the existing constructors of `DBLog` are all agnostic to the type of database.

OTOH, I think your suggestion would make sense if we had:

- `PoolDbLog`
- `WalletDbLog`

Reply:
I've created a technical debt ticket to cover the idea of defining separate `WalletDbLog` and `PoolDbLog` types here: https://jira.iohk.io/browse/ADP-407.

While tackling the above ticket, we could also go through the existing `DbLog` constructors and replace the "stringly-typed" constructors with strongly-typed equivalents in `WalletDbLog` and `PoolDbLog`. Hopefully, this should also clean up parts of the code that are obscured with log formatting logic.

Reply:
I would like this to be fixed now, rather than kicking the can down the road.

Reply:
@rvl wrote:
> I would like this to be fixed now, rather than kicking the can down the road.

I disagree that fixing this was an _essential_ part of _this_ PR.

I think it's generally better that a PR has just one clear function, as smaller, self-contained PRs are easier to review, and make it easier to avoid scope creep.

In addition, we already had a ticket in place to perform the type separation. It would have been a simple matter to make a follow-up PR from that ticket.

Nevertheless, I've added the following commits:

* 97f3901c48d93eece61775337091afa3f8d166f1 (Provide database-schema-specific logging types.)
* b3c96732a6a9a778d3a40f533b7ec87d53b70aa1 (Introduce log constructor `MsgRemovingPool`.)
* 15a67a3d2e0be6813c56d35ac28c0331d3520e46 (Use `MsgRemovingPool` in `removePools` operation.)
* 0741d66c861d7589c0ba8d982cbe07023e1fce8d (Remove unused `MsgRemovingDatabaseEntity` log constructor.)

Could have a look at these commits and let me know your thoughts?

Thanks!